content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -275,33 +275,79 @@ node {
 }
 ----
 
-=== Defining steps
 
+=== Defining global variables
+
+Internally, scripts in the `vars` directory are instantiated as a singleton
+on-demand, when used first.  So it is possible to define more methods,
+properties on a single file that interact with each other:
+
+[source,groovy]
+----
+// vars/acme.groovy
+def setFoo(v) {
+    this.foo = v;
+}
+def getFoo() {
+    return this.foo;
+}
+def say(name) {
+    echo "Hello world, ${name}"
+}
+----
+
+Then the Pipeline Script can invoke these methods, rooted off the `acme`
+object:
+
+[source,groovy]
+----
+acme.foo = "5";
+echo acme.foo; // print 5
+acme.say "Joe" // print "Hello world, Joe"
+----
+
+[NOTE]
+====
+A variable defined in a shared library will only show up in _Global Variables
+Reference_ (under _Pipeline Syntax_) after you have first run a successful
+build using that library, allowing its sources to be checked out by Jenkins.
+====
 
-Shared Libraries can also define functions which look and feel like built-in
-steps, such as `sh` or `git`.
+=== Defining steps
 
+Shared Libraries can also define global variables which look and feel like
+built-in steps, such as `sh` or `git`.
 
-For example, to define `helloWorld` step, the file `vars/helloWorld.groovy`
-should be created and have a `call` method defined:
+For example, to define `helloWorld`, the file `vars/helloWorld.groovy`
+should be created and should implement a `call` method. The `call` method
+allows the global variable to be invoked in a manner similar to an step:
 
 [source,groovy]
 ----
 // vars/helloWorld.groovy
 def call(name) {
-    // you can call any valid step functions from your code, just like you can from Pipeline scripts
+    // Any valid steps can be called from this code, just like in other
+    // Pipeline scripts
     echo "Hello world, ${name}"
 }
 ----
 
-The Pipeline Script would then be able to call this step:
+The Pipeline Script would then be able to reference and invoke this variable:
 
 [source,groovy]
 ----
 helloWorld "Joe"
 ----
 
-If called with a block, the `call` method will receive a 
+[NOTE]
+====
+The `call()` method will not be automatically invoked if `helloWorld` is
+referenced without *any* arguments. Implementing a global variable which
+requires no default arguments **must** be explicitly invoked like a method, for
```

Comment:
I think the updated example you suggest @bitwiseman, with an inline code comment perhaps, does a decent job conveying things here.

I quite dislike the tone I'm perceiving in this review here though. There was a stumbling block which I hit (relatively savvy Pipeline user) and at least other user hit, hard enough to file a ticket. 

Avoiding referencing or spelling out how "basic groovy language facts" work in Pipeline is a slightly more dressed up version of "lrn2groovy noob" in my opinion.

If users are hitting stumbling blocks, or having problems with Pipeline Script, that is a *bug* to me. Whether it's fixed in the UI, documentation, or Pipeline's syntax, something should be done to help future users avoid the same potholes.



That said, @bitwiseman are you planning on submitting a subsequent PR, or should I incorporate your updated examples here?